File: 589.java
class Solution {
    private void helper(Node root, List<Integer> output){
        if(root == null) return;
        output.add(root.val);
        if(root.children != null){
            for(Node child : root.children) 
                helper(child, output);
        }
    }
    public List<Integer> preorder(Node root) {
        List<Integer> output = new ArrayList<>();
        helper(root, output);
        return output;
    }
}

File: 589.py
class Solution:
    def preorder(self, root: 'Node') -> List[int]:
        
        
        stack=collections.deque()
        stack.append(root)
        result=[]
        while(stack):
            curr=stack.popleft()
            if(curr):
                result.append(curr.val)
                [stack.appendleft(child) for child in curr.children[::-1]]
        return result